owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/users/GetRemoteUserAvatarOperation.kt

```diff
@@ -0,0 +1,109 @@
+/* ownCloud Android Library is available under MIT license
+ *
+ *   Copyright (C) 2020 ownCloud GmbH.
+ *
+ *   Permission is hereby granted, free of charge, to any person obtaining a copy
+ *   of this software and associated documentation files (the "Software"), to deal
+ *   in the Software without restriction, including without limitation the rights
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+package com.owncloud.android.lib.resources.users
+
+import com.owncloud.android.lib.common.OwnCloudClient
+import com.owncloud.android.lib.common.http.HttpConstants
+import com.owncloud.android.lib.common.http.methods.nonwebdav.GetMethod
+import com.owncloud.android.lib.common.network.WebdavUtils
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import timber.log.Timber
+import java.io.IOException
+import java.io.InputStream
+import java.net.URL
+
+class GetRemoteUserAvatarOperation(private val avatarDimension: Int) :
+    RemoteOperation<RemoteAvatarData>() {
+    override fun run(client: OwnCloudClient): RemoteOperationResult<RemoteAvatarData> {
+        var inputStream: InputStream? = null
+        lateinit var getMethod: GetMethod
+
+        lateinit var result: RemoteOperationResult<RemoteAvatarData>
+        try {
+            val endPoint =
+                client.baseUri.toString() + NON_OFFICIAL_AVATAR_PATH + client.credentials.username + "/" + avatarDimension
+            Timber.d("avatar URI: %s", endPoint)
+
+            getMethod = GetMethod(URL(endPoint))
+
+            val status = client.executeHttpMethod(getMethod)
+
+            if (isSuccess(status)) {
+                // find out size of file to read
+                val contentLength = getMethod.getResponseHeader(HttpConstants.CONTENT_LENGTH_HEADER).toInt()
+
+                // find out MIME-type!
+                val mimeType = getMethod.getResponseHeader(HttpConstants.CONTENT_TYPE_HEADER)
+
+                if (mimeType == null || !mimeType.startsWith("image")) {
```

Comment:
I would use a constant for "image" or even a String extension to check mimeTypes => mimeType.isImageType()

Reply:
That will be done in files rearquitecture.